### ISSUE TITLE
incident: bokeh server running but 'figure' app shows blank page

### ISSUE THREAD

**Issue by ltalirz** [ltalirz/lsmo-bokeh-app#24]
as reported by @giovannipizzi, the figure app at https://discover.materialscloud.org/cofs/figure was found in a state where nothing was displayed (white page).
The javascript console showed no errors.

The issue was resolved by restarting the docker container.

The docker logs contain only one type of error (96 times in total):
```
2018-10-31 13:24:27,117 Error running application handler <bokeh.application.handlers.directory.DirectoryHandler object at 0x7fbecdf1c6d0>: '3aasdad'
File "main.py", line 40, in load_preset:
preset = copy(presets[new]) Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/bokeh/application/handlers/code_runner.py", line 163, in run
    exec(self._code, module.__dict__)
  File "/project/discover-cofs/figure/main.py", line 152, in <module>
    load_preset(None, None, get_preset_label_from_url())
  File "/project/discover-cofs/figure/main.py", line 40, in load_preset
    preset = copy(presets[new])
KeyError: '3aasdad'
```
(i.e. '3aasdad' is passed as the preset string via a variable in the url, which causes the app to fail).

Need to check where this is coming from.

**Comment by ltalirz** [comment]
* the string `3aasdad` was found nowhere in the codebase using
`git grep 3aasdad $(git rev-list --all)`. Unclear where this came from
 * wrapper the selection of preset in try/except to prevent this error in the future 3921a20b1c1d2fd036306a6ef75e025ba3e34934

The fundamental reason for the blank page being displayed is unclear to me (bokeh server still seemed to be responding).
Perhaps preventing the error from occurring will fix this?

**Comment by ltalirz** [comment]
It's actually caused by an issue in bokeh that results in any exception killing the bokeh app for all connected users. https://github.com/bokeh/bokeh/issues/8034
This issue was closed incorrectly but persists.

As long as it does, the suggestion is to simply catch all possible exceptions that might occur.

**Comment by ltalirz** [comment]
fixed in 760a4927fa58934e37e3b15c86276428c2484c45 by moving to bokeh 1.2.0
